components/EditDatesForm.js

```diff
@@ -0,0 +1,86 @@
+import styled from "styled-components";
+import Image from "next/image";
+import saveSvg from "../assets/save.svg";
+import { useState } from "react";
+
+export default function EditDatesForm({ destination, onSubmitChangeDetail, toggleModal }) {
+  const timestampToIsoString = (timestamp) => new Date(timestamp * 1000).toISOString().substring(0, 10);
+
+  const [startDate, setStartDate] = useState(timestampToIsoString(destination.startDate));
+  const [endDate, setEndDate] = useState(timestampToIsoString(destination.endDate));
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    onSubmitChangeDetail({
+      changed: "Dates",
+      startDate: Date.parse(startDate) / 1000,
+      endDate: Date.parse(endDate) / 1000,
+    });
```

Comment:
You are constructing an object with a changed property. That is unnecessary. You can just create an object which contains an arbitrary number of changed properties, like so:
```
{
      startDate: 1668902400,
      endDate: 1670112000,
      hotel: "Lebua State Hotel"
}
```

In your onSubmitChangeDetail function you can simply write:
```
return {
            ...destinationItem,
            ...changedData,
          }
```
This way your onSubmitChangeDetail will get easier to understand and reusable. You can just feed it any changed data and will work.

Reply:
Oh thanks! I actually was about to ask you about that.

Reply:
Done.